echoregions/lines/lines_mask.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+import xarray as xr
+from xarray import DataArray
+
+from .lines import Lines
+
+
+def lines_mask(
+    da_Sv: DataArray, lines: Lines, method: str = "nearest", limit_area: str = None
+):
+    """
+    Subsets a bottom dataset to the range of an Sv dataset
+    Create a mask of same shape as data found in Sonar object; bottom: False, otherwise: True
+
+    Arguments:
+        sonar - Sonar object containing DataArray of shape (ping_time, depth).
+        lines - Lines object containing bottom values.
+        method - String containing interpolation method.
+        limit_area - String for determining filling restriction for NA values.
+
+    Returns:
+      bottom_mask - xarray with dimensions: (ping_time, depth) with bottom: False, otherwise: True
+    """
+
+    def filter_bottom(bottom, start_date, end_date):
+        """
+        Selects the values of the bottom between two dates.
+        """
+        after_start_date = bottom["time"] > start_date
+        before_end_date = bottom["time"] < end_date
+        between_two_dates = after_start_date & before_end_date
+        filtered_bottom = bottom.loc[between_two_dates].set_index("time")
+        return filtered_bottom
+
+    if type(lines) != Lines:
+        raise TypeError(
+            f"lines should be of type Lines. lines is currently of type {type(lines)}."
+        )
+
+    lines_df = lines.data
+
+    # new index
+    sonar_index = list(da_Sv.ping_time.data)
+
+    # filter bottom within start and end time
+    start_time = da_Sv.ping_time.data.min()
+    end_time = da_Sv.ping_time.data.max()
+    filtered_bottom = filter_bottom(lines_df, start_time, end_time)
+
+    if len(filtered_bottom) > 0:
+        # create joint index
+        joint_index = list(
+            set(list(pd.DataFrame(sonar_index)[0]) + list(filtered_bottom.index))
+        )
+
+        # interpolate on the sonar coordinates
+        # nearest interpolation has a problem when points are far from each other
+        bottom_interpolated = filtered_bottom.reindex(joint_index).loc[
+            sonar_index
+        ]  # .interpolate('nearest').loc[sonar_index]
```

Comment:
I think one can do reindex directly to `sonar_index` without creating the `joint_index` first?